arbnode/dataposter/data_poster.go

```diff
@@ -1169,13 +1183,27 @@ func (p *DataPoster) Start(ctxIn context.Context) {
 		if latestQueued != nil {
 			latestCumulativeWeight = latestQueued.CumulativeWeight()
 			latestNonce = latestQueued.FullTx.Nonce()
+
+			var confirmedWeight uint64
+			confirmedNonce := unconfirmedNonce - 1
+			confirmedMeta, err := p.queue.Get(ctx, confirmedNonce)
+			if err != nil {
+				log.Error("Failed to fetxh latest confirmed tx from queue", "err", err)
+				return minWait
```

Comment:
We shouldn't return an error here. Confirmed txs might be missing, e.g. if we switch data poster redis instances.

Reply:
I think simply logging a warning and not updating the metrics in this case would be good